homeassistant/components/plant.py

```diff
@@ -171,15 +171,15 @@ def state_changed(self, entity_id, _, new_state):
 
         reading = self._sensormap[entity_id]
         if reading == READING_MOISTURE:
-            self._moisture = int(value)
+            self._moisture = int(float(value))
```

Comment:
Why not stay with the float and round it? Or do the same as we do in the `climate` component with `PRECISION_HALVES` and the others?

Reply:
I'm not sure if there was a specific reason to use int in the first place, or if changing the type would have some other adverse effects. While I don't see a reason against your suggestion, this change is the smallest one I could think of to fix the problem and int is fine for me.